dashboard/app/helpers/thank_donors_interstitial_helper.rb

```diff
@@ -8,6 +8,8 @@ def self.show?(user, request)
     return false if user.age.nil? # Age dialog
     return false if user && user.teacher? && !user.accepted_latest_terms? # Terms of Service dialog
     return false if GdprDialogHelper.show?(user, request)
+    return false if SchoolInfoInterstitialHelper.show?(user)
+    return false if SchoolInfoInterstitialHelper.show_confirmation_dialog?(user)
```

Comment:
This approach looks good but these method have [side effects](https://github.com/code-dot-org/code-dot-org/blob/da7adceb01591ec1077fdf21a30f5c575b891b83/dashboard/app/helpers/school_info_interstitial_helper.rb#L21) 😭 

So we'd either have to refactor those (which tbh we probably should, I'm pretty anti these methods using the `?` syntax and having side effects, but that might not be necessary for this PR) or we could move this logic to [index.html.haml](https://github.com/code-dot-org/code-dot-org/blob/da7adceb01591ec1077fdf21a30f5c575b891b83/dashboard/app/views/home/index.html.haml#L22). I wonder if there's a way to enforce that we only show one dialog in that file

Reply:
Ahh, I obviously did not look closely at that, thank you for catching!

I think I was sort of envisioning we move towards a world where every dialog has a show? helper (Elijah suggested these should actually be "policy objects", but I think that's the same idea) -- I refactored the GDPR logic in the donor dialog work to have a helper, and obviously added one for whether to show the donor dialog. Then we have some central place that prefs which to show (maybe index.html.haml, or another helper)?

Reply:
Could I move the user update to index.html.haml (instead of in the show? method)?

Reply:
Gave it a shot, LMK what you think!

Reply:
The new update looks good to me! I'd be surprised (and disappointed) if this doesn't end up breaking tests, so those might need to be modified

Reply:
I did check the helper test file and didn't see anything that looked like it would fail :|

Reply:
Yeah, this passes Drone. That's disappointing :(

Reply:
It looks like we once had a [test](https://github.com/code-dot-org/code-dot-org/blob/c8ecf1aee7d845bac8cc80e257e04e89cdcb5fb2/dashboard/test/lib/school_info_interstitial_helper_test.rb#L104) for this but I can't figure out when it was removed...